utilities/roswtf/src/roswtf/network.py

```diff
@@ -49,7 +49,8 @@ def ip_check(ctx):
     local_addrs = rosgraph.network.get_local_addresses()
 
     resolved_ips = [host[4][0] for host in socket.getaddrinfo(socket.gethostname(), 0, 0, 0, socket.SOL_TCP)]
-    global_ips = [ ip for ip in resolved_ips if not ip.startswith('127.') and not ip == '::1']
+    # IPv6 support % to denote zone/scope ids, that is why the replace commands are in place
+    global_ips = [ ip for ip in resolved_ips if not ip.replace('%','%%').startswith('127.') and not ip.replace('%','%%') == '::1']
```

Comment:
Since these variables are only used in one spot you could just change the substitution syntax below to use `str.format` style substitution (also adding a comment as to why using `.format` is important). Then you would not need to escape anything.